src/runtime_src/core/include/xrt/xrt_pskernel.h

```diff
@@ -0,0 +1,713 @@
+/*
+ * Copyright (C) 2021, Xilinx Inc - All rights reserved
+ * Xilinx Runtime (XRT) Experimental APIs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef _XRT_PSKERNEL_H_
+#define _XRT_PSKERNEL_H_
+
+#include "ert.h"
+#include "xrt.h"
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_uuid.h"
+
+#ifdef __cplusplus
+# include "experimental/xrt_enqueue.h"
+# include <chrono>
+# include <cstdint>
+# include <functional>
+# include <memory>
+# include <vector>
+#endif
+
+/**
+ * typedef xrtPSKernelHandle - opaque kernel handle
+ *
+ * A kernel handle is obtained by opening a kernel.  Clients
+ * pass this kernel handle to APIs that operate on a kernel.
+ */
+typedef void * xrtPSKernelHandle;
+
+/**
+ * typedef xrtPSRunHandle - opaque handle to a specific kernel run
+ *
+ * A run handle is obtained by running a kernel.  Clients
+ * use a run handle to check or wait for kernel completion.
+ */
+typedef void * xrtPSRunHandle;  // NOLINT
+
+#ifdef __cplusplus
+
+namespace xrt {
+
+class pskernel;
+class event_impl;
+
+/*!
+ * @class psrun 
+ *
+ * @brief 
+ * xrt::run represents one execution of a kernel
+ *
+ * @details
+ * The run handle can be explicitly constructed from a kernel object
+ * or implicitly constructed from starting a kernel execution.
+ *
+ * A run handle can be re-used to execute the same kernel again.
+ */
+class psrun_impl;
+class psrun
```

Comment:
```suggestion
class ps_run_impl;
class ps_run
```
More generally, would it make sense to have a `xrt::ps` if there are `run`, `kernel`...